### DISCUSSION TITLE
Could not activate copilot in online vscode provider

### DISCUSSION THREAD

**Discussion by katorly** [community/community#10674] [Copilot Conversations]
I have just got access for [Github Copilot](https://github.com/features/copilot/signup) early this morning, and I eagerly installed it on an online Visual Studio Code provider. I had my Github account logged in and the extension installed correctly, but its output just kept saying:
```console
[INFO] [auth] [2022-01-28T04:30:35.865Z] Invalid copilot token: missing token: 404 
[ERROR] [default] [2022-01-28T04:30:35.867Z] GitHub Copilot could not connect to server. Extension activation failed: "User not authorized"
```
and I can't use any of its features.  

What I'm sure is:  
- I have access to Copilot Technical Preview.
- Copilot can run normally in my local vscode.
- That online vscode is a thing **like** Codespaces, **NOT** vscode.dev.
- The copilot icon did appeared, showing it disabled.
- Both me and the online provider‘s Internet connection is OK.

**Comment by AcmeGamers** [comment]
I believe the reason would be it is not for the web at the moment. As you can see in the image

![image](https://user-images.githubusercontent.com/52010365/151651070-241fc6ac-5b67-4cb3-9cdd-ace10d65dff3.png)

**Reply by katorly** [reply]
What I've said in the post:
> That online vscode provider isn't such thing like vscode.dev, which has a lot limitations.

![pic](https://user-images.githubusercontent.com/70022443/151652134-8c55c445-0c27-4cf8-811f-d643671e06be.png)

**Reply by AcmeGamers** [reply]
Mhm, although do you want to access it on your VS Code online? (Which I'm sure you are aware of this issue)

**Reply by katorly** [reply]
> Mhm, although do you want to access it on your VS Code online? (Which I'm sure you are aware of this issue)

That "VS Code online" is a thing like Codespaces, not vscode.dev

**Reply by AcmeGamers** [reply]
Ah, my bad, just saw the title is vscode provider. If that's the case, try re-logging with GitHub on the service provider you are using (disconnect it and then login again), it might help

**Reply by katorly** [reply]
> Ah, my bad, just saw the title is vscode provider. If that's the case, try re-logging with GitHub on the service provider you are using (disconnect it and then login again), it might help

Sorry, but that didn't help... The Copilot kept saying `could not connect to server`

**Reply by yourlocaltechboi** [reply]
I can verify this issue is present on Gitpod.

**Reply by katorly** [reply]
> I can verify this issue is present on Gitpod.

Yes it is.

**Comment by techopswhiz** [comment]
Ask your service provider to whitelist outbound connections to copilot, it's probably blocked in their network.
